shell.nix

```diff
@@ -3,25 +3,26 @@
       fetchTarball
         {
           name = "21.11";
```

Comment:
Done!

I set to to `nixpkgs-${release}-${date}` so that both `getName` and `getVersion` returns meaningful values:
```
nix-repl> lib.getName "nixpkgs-21.11-2022-01-03"
"nixpkgs"

nix-repl> lib.getVersion "nixpkgs-21.11-2022-01-03"
"21.11-2022-01-03"
```